styx-common/src/test/java/com/spotify/styx/util/WorkflowValidatorTest.java

```diff
@@ -247,6 +248,30 @@ public void shouldNotAllowComponentWithHash(String component) {
         contains("component id cannot contain #"));
   }
 
+  @Test
+  public void shouldFailUsageOfNonWhitelistedSecret() {
+    WorkflowValidator sut = WorkflowValidator.newBuilder(dockerImageValidator)
+        .withSecretWhitelist(Set.of("bar-secret"))
+        .build();
+
+    final List<String> errors = sut.validateWorkflow(
+        Workflow.create("test", FULL_WORKFLOW_CONFIGURATION));
+
+    assertThat(errors, contains("secret " + FULL_WORKFLOW_CONFIGURATION.secret().get().name() + " is not whitelisted"));
+  }
+
+  @Test
+  public void shouldPassUsageOfWhitelistedSecret() {
+    WorkflowValidator sut = WorkflowValidator.newBuilder(dockerImageValidator)
+        .withSecretWhitelist(Set.of(FULL_WORKFLOW_CONFIGURATION.secret().get().name()))
+        .build();
+
+    final List<String> errors = sut.validateWorkflow(
+        Workflow.create("test", FULL_WORKFLOW_CONFIGURATION));
+
+    assertThat(errors, empty());
+  }
+
```

Comment:
Can you add also a test named like `shouldPassUsageForEmptyWhitelist` to verify that we wont fail if there is no white list?

Reply:
I think this is covered by all the other test cases where we use `sut`.

Reply:
Agree